### PR TITLE
Kafka producer changes

### PR DESCRIPTION
**Pull request by jrheizelman**
Added batching to the kafka producer using the same value given for CPS.

### CODE REVIEW

File: load-test-framework/src/main/java/com/google/pubsub/clients/gcloud/CPSPublisherTask.java [hunk 1]
```diff
@@ -52,10 +54,9 @@ private CPSPublisherTask(String project, String topic, int messageSize, int batc
   }
 
   public static void main(String[] args) throws Exception {
-    LoadTestRunner.run(request ->
+    LoadTestRunner.run((Function<StartRequest, Task>) request ->
```
Comment (mdietz94) [inline load-test-framework/src/main/java/com/google/pubsub/clients/gcloud/CPSPublisherTask.java]: I don't think you need this cast.

File: load-test-framework/src/main/java/com/google/pubsub/clients/gcloud/CPSSubscriberTask.java [hunk 2]
```diff
@@ -47,7 +49,7 @@ private CPSSubscriberTask(String project, String subscription, int batchSize) {
   }
 
   public static void main(String[] args) throws Exception {
-    LoadTestRunner.run(request ->
+    LoadTestRunner.run((Function<StartRequest, Task>) request ->
```
Comment (mdietz94) [inline load-test-framework/src/main/java/com/google/pubsub/clients/gcloud/CPSSubscriberTask.java]: I also don't think you need this cast.

File: load-test-framework/src/main/java/com/google/pubsub/clients/kafka/KafkaPublisherTask.java [hunk 3]
```diff
@@ -37,47 +40,49 @@
...
+    this.batchSize = batchSize;
     Properties props = new Properties();
-    props.putAll(ImmutableMap.of(
-        "max.block.ms", "30000",
-        "key.serializer", "org.apache.kafka.common.serialization.StringSerializer",
-        "value.serializer", "org.apache.kafka.common.serialization.StringSerializer",
-        "acks", "all",
-        "bootstrap.servers", broker
-    ));
+    props.putAll(new ImmutableMap.Builder<String, String>().
```
Comment (mdietz94) [inline load-test-framework/src/main/java/com/google/pubsub/clients/kafka/KafkaPublisherTask.java]: I don't think you need <String, String>, in general use <> when possible.

File: load-test-framework/src/main/java/com/google/pubsub/clients/kafka/KafkaSubscriberTask.java [hunk 4]
```diff
@@ -52,7 +54,7 @@ private KafkaSubscriberTask(String broker, String project, String topic, long po
   }
 
   public static void main(String[] args) throws Exception {
-    LoadTestRunner.run(request ->
+    LoadTestRunner.run((Function<StartRequest, Task>) request ->
```
Comment (mdietz94) [inline load-test-framework/src/main/java/com/google/pubsub/clients/kafka/KafkaSubscriberTask.java]: Don't need cast

File: load-test-framework/src/main/java/com/google/pubsub/clients/gcloud/CPSPublisherTask.java (hunk 1 above)
Comment (jrheizelman) [inline load-test-framework/src/main/java/com/google/pubsub/clients/gcloud/CPSPublisherTask.java]: This isn't necessary for the compiler, but I added it 1. to silence an Eclipse warning, 2. I think it makes it more readable when you know what request is without having to go to LoadTestRunner to check.

File: load-test-framework/src/main/java/com/google/pubsub/clients/kafka/KafkaPublisherTask.java [hunk 5]
```diff
@@ -37,47 +40,49 @@
...
+        put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer").
+        put("acks", "all").
+        put("bootstrap.servers", broker).
+        put("batch.size", Integer.toString(batchSize)).build());
     this.publisher = new KafkaProducer<>(props);
   }
 
   public static void main(String[] args) throws Exception {
-    LoadTestRunner.run(request ->
+    LoadTestRunner.run((Function<StartRequest, Task>) request ->
```
Comment (jrheizelman) [inline load-test-framework/src/main/java/com/google/pubsub/clients/kafka/KafkaPublisherTask.java]: Ditto

File: load-test-framework/src/main/java/com/google/pubsub/clients/kafka/KafkaSubscriberTask.java (hunk 4 above)
Comment (jrheizelman) [inline load-test-framework/src/main/java/com/google/pubsub/clients/kafka/KafkaSubscriberTask.java]: Ditto

File: load-test-framework/src/main/java/com/google/pubsub/clients/kafka/KafkaPublisherTask.java (hunk 3 above)
Comment (jrheizelman) [inline load-test-framework/src/main/java/com/google/pubsub/clients/kafka/KafkaPublisherTask.java]: Fixed

**Comment by jrheizelman** [comment]
Made all the changes with the exception of the unnecessary cast. Let me know what your thoughts are on the readability issue I raised and I'll remove them if you still think I should.

Also, the travis test failed bc it didn't have permissions for run.py? Did anything get changes in the yaml file that could cause that?

**Comment by mdietz94** [comment]
I'd really prefer you remove the unnecessary casts, but if you push back
again I'll LGTM. And yes, I fixed this in my other branch, I think I may
have done it inadvertantly

On Mon, Oct 24, 2016, 6:07 PM John Heizelman <notifications@github.com>
wrote:

Made all the changes with the exception of the unnecessary cast. Let me
know what your thoughts are on the readability issue I raised and I'll
remove them if you still think I should.

Also, the travis test failed bc it didn't have permissions for run.py? Did
anything get changes in the yaml file that could cause that?

—
You are receiving this because you commented.


Reply to this email directly, view it on GitHub
<https://github.com/GoogleCloudPlatform/pubsub/pull/17#issuecomment-255880332>,
or mute the thread

File: load-test-framework/src/main/java/com/google/pubsub/clients/common/MetricsHandler.java [hunk 6]
```diff
@@ -178,9 +178,13 @@ private void createMetrics() {
       log.info("Metrics already exist.");
     }
   }
-
+  
   public synchronized void recordLatency(long latencyMs) {
```
Comment (mdietz94) [inline load-test-framework/src/main/java/com/google/pubsub/clients/common/MetricsHandler.java]: We do not use operator overloading wherever possible. Please undo this. It was done like that to also avoid from grabbing and releasing the lock as often when recording a batch.

**Comment by jrheizelman** [comment]
I moved the throughput stat tracking changes to a new branch, so this is now back where it was when you approved the changes plus a one-line change removing an unnecessary call to the kafka consumer. I'll fix up the issues with the new stuff (it wasn't ready and I was planning on moving the commits out before merging anyways) and submit another PR.

**Pull request merged**